lib/init/grass.py

```diff
@@ -1900,10 +1900,21 @@ def csh_startup(location, location_name, mapset, grass_env_file):
     os.environ['HOME'] = userhome
     return process
 
+def shell_startup(location, location_name, grass_env_file, shell_name):
+    if shell_name not in ['bash', 'zsh']:
```

Comment:
This case would be a programmer's mistake. Since this function supports `msh` and `cygwin` as well (maybe more in the future) and the user cannot make this error, I wouldn't check for `shell_name` here.

Reply:
> This case would be a programmer's mistake.

This was exactly my point, with the function naming problem this was a way to avoid potential errors. But if we go with the solution of moving the handling of the different shells, this is indeed superfluous.

Reply:
I reorganised the initial shell-check, but I think it doesn't hurt in this case to throw an exception if a none-supported shell name is called.